File: csharp-solution/E-CommerceSystem/Models/ExpirableProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace E_CommerceSystem.Models
{
  public class ExpirableProduct:Product
  {
    public ExpirableProduct(string name, double price, int quantiy,DateTime expiryDate):base(name,price,quantiy)
    {
      ExpiryDate = expiryDate;
    }

    private DateTime ExpiryDate;
    public override bool IsExpired()
    {
      return ExpiryDate < DateTime.Now.Date;
    }

  }
}

File: java-solution/src/Model/ExpirableProduct.java
package Model;

import interfaces.IExpirable;

import java.time.LocalDate;

public class ExpirableProduct extends Product implements IExpirable {
    private LocalDate expiryDate;
    public ExpirableProduct(String name, double price, int quantity,LocalDate expiryDate) {
        super(name, price, quantity);
        this.expiryDate=expiryDate;
    }

    public boolean isExpired() {
        return LocalDate.now().isAfter(expiryDate);
    }
}
